src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -88,6 +88,89 @@ class LinearIsotropicConductor {
   double conductivity_;
 };
 
+/// Linear isotropic thermal conduction material model
+class ParameterizedLinearIsotropicConductor {
+public:
+  /**
+   * @brief Construct a new Parameterized Linear Isotropic Conductor object
+   *
+   * @param density Density of the material (mass/volume)
+   * @param specific_heat_capacity Specific heat capacity of the material (energy / (mass * temp))
+   * @param conductivity Thermal conductivity of the material (power / (length * temp))
+   */
+  ParameterizedLinearIsotropicConductor(double density = 1.0, double specific_heat_capacity = 1.0,
+                                        double conductivity = 1.0)
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_(conductivity)
+  {
+    SLIC_ERROR_ROOT_IF(conductivity_ < 0.0,
+                       "Conductivity must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic conductor material model.");
+
+    SLIC_ERROR_ROOT_IF(specific_heat_capacity_ < 0.0,
+                       "Specific heat capacity must be positive in the linear isotropic conductor material model.");
+  }
+
+  /**
+   * @brief Function defining the thermal flux (constitutive response)
+   *
+   * @tparam T1 type of the temperature (e.g. tensor or dual type)
+   * @tparam T2 type of the temperature gradient (e.g. tensor or dual type)
+   * @tparam T3 type of the parameter field (e.g. tensor or dual type)
+   * @param temperature_gradient Gradient of the temperature (du_dx)
+   * @param parameter The user-defined parameter used to calculate the thermal conductivity
+   * @return The thermal flux of the material model
+   */
+  template <typename T1, typename T2, typename T3>
+  SERAC_HOST_DEVICE auto operator()(const T1& /* temperature */, const T2& temperature_gradient,
```

Comment:
Can `maybe_unused` be used with arguments like temperature?

Reply:
It would be syntactically okay to decorate function parameters with `[[maybe_unused]]`, but I don't think it makes sense in this particular case, since `temperature` is _always_ unused here-- there's no "maybe" about it!

Reply:
Yeah, I like doing this because it shows explicitly that it is unused.

Reply:
Would `[[maybe_unused]]` work with Doxygen?

Reply:
A good question! I'll look into this and we can discuss it in the weekly meeting.